### ISSUE TITLE
Disbale compiler instead of aborting if kernel library is not found at runtime

### ISSUE THREAD

**Issue by Oblomov** [pocl/pocl#1346]
While playing around with HSA device to make it work on my hardware, I noticed that `getKernelLibrary` calls `POCL_ABORT` if the kernel library is not found.

I think a more robust approach would be the following:

* if the device supports compilation and/or linking, `CL_DEVICE_COMPILER_AVAILABLE` and `CL_DEVICE_LINKER_AVAILABLE` start in a “unknown” state rather than `CL_TRUE`;
* the accessibility of the kernel library is checked at run time under one of these conditions:
    * the user program queries for the `CL_DEVICE_COMPILER_AVAILABLE` or `CL_DEVICE_LINKER_AVAILABLE` properties;
    * the user program calls `clBuildProgram` or `clLinkProgram`;
* if the kernel library _cannot_ be accessed, the properties are set to `CL_FALSE` and an appropriate error is returned for `clBuildProgram` and `clLinkProgram`;

**Comment by pjaaskel** [comment]
Not sure of this because isn't it a case of a broken PoCL or a PoCL installation? If it cannot find the bitcode library that is.

**Comment by franz** [comment]
I agree that it should not abort, however IMO the proposed solution is too over-complicated with very little payoff.... Should be sufficient to return a failure from inside the build-program / link-program call, plus properly inform the user via the log.

Also, reading the `clGetDeviceInfo` documentation on `CL_DEVICE_COMPILER_AVAILABLE`:

```
Is CL_TRUE if the compiler is available. This can be CL_FALSE for the embedded platform profile only.
```

... so it seems to me that we actually cannot set the `CL_DEVICE_COMPILER_AVAILABLE` to anything but `CL_TRUE`, except in a few drivers like ALMAIF and Remote.

**Comment by pjaaskel** [comment]
Even Remote does support online compilation (remotely).

**Comment by Oblomov** [comment]
> Not sure of this because isn't it a case of a broken PoCL or a PoCL installation? If it cannot find the bitcode library that is.

Not necessarily broken, although it probably does indicate some dependency issue. Either way, I believe graceful degradation is preferable to an abort.

FWIW, if I'm reading the CUDA device correctly it's already implementing at least the “set compiler as unavailable in the device info”.

**Comment by pjaaskel** [comment]
Please send a PR instead if implementing this.
